### ISSUE TITLE
Error in parsing

### ISSUE THREAD

**Issue by dsrich** [faassen/xot#3]
I used the example code from the readme, added a couple of unwrap()s to make it compile, and it returns the error below on what appears to be a perfectly formed XML entry. I include the error message and XML below. This is Rust 2021, fully up to date as of 23 June 2023. I am guessing the ampersand character is causing the problem.

The code:
```rust
    let xml_bk = std::fs::read_to_string("Appal.xml").unwrap();
    let mut xot = Xot::new();
    let root = xot.parse(&xml_bk).unwrap();
    let doc_el = xot.document_element(root).unwrap();
    let txt = xot.first_child(doc_el).unwrap();
    let txt_value = xot.text(txt).unwrap().get().clone();
```
Error Message:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: UnclosedEntity(" Crystal prisoner\n -   Where did they come from and\n -   How did Aaron & Crystal not know about them before they left?\n -   Ed and Harry had been surveilling every troll movement since? Could they have missed?\n -   Not tracking small groups, but how small is small? \n \n Aaron jumps in bed w/Gayle a month after he is back \n -   Rebound relationship is understood but they are rarely stable\n -   At a minimum, they have some \"growing together\" needed-their personalities are wildly divergent\n -   Or are they? Does she appeal to the alpha rebel underneath his more civilized shell, something he isn't aware of?")', src/main.rs:15:35
```
The XML surrounding that is:
```xml
    <PROJECTNOTE>
      <ID>5</ID>
      <Title><![CDATA[Note from Timeline exercise 20210729]]></Title>
      <Desc><![CDATA[ The first group of Trolls that took Aaron & Crystal prisoner
 -   Where did they come from and
 -   How did Aaron & Crystal not know about them before they left?
 -   Ed and Harry had been surveilling every troll movement since? Could they have missed?
 -   Not tracking small groups, but how small is small? 
 
 Aaron jumps in bed w/Gayle a month after he is back 
 -   Rebound relationship is understood but they are rarely stable
 -   At a minimum, they have some "growing together" needed-their personalities are wildly divergent
 -   Or are they? Does she appeal to the alpha rebel underneath his more civilized shell, something he isn't aware of?]]></Desc>
      <SortOrder>5</SortOrder>
    </PROJECTNOTE>
```

**Comment by dsrich** [comment]
Confirmed with test data:
```rust
fn main() {

    let xml_bk = r#"<tag1 att1 = "test">
        <tag2><!--Test comment-->Test</tag2>
        <tag2>Test & 2</tag2>
     </tag1>"#;

    let mut xot = Xot::new();
    let root = xot.parse(&xml_bk).unwrap();
    let doc_el = xot.document_element(root).unwrap();
    let txt = xot.first_child(doc_el).unwrap();
    let txt_value = xot.text(txt).unwrap().get().clone();

    println!("{}", txt_value);
}
```
Error
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: UnclosedEntity(" 2")', src/main.rs:15:35
```

**Comment by faassen** [comment]
Thanks for the report! Clearly I need to test with more diverse XML. I will take a look on Monday.

Would also appreciate a fix to the doc example.

**Comment by dsrich** [comment]
The doc example problems were because I didn't have a return value from main(), which meant the question marks wouldn't work, and I had to use unwrap() instead. I don't often return a result from main().

**Comment by faassen** [comment]
Hm, I've been looking at the XML specification about whether `&` is legal:

https://www.w3.org/TR/xml/#syntax

says:

> The ampersand character (&) and the left angle bracket (<) MUST NOT appear in their literal form, except when used as 
> markup delimiters, or within a [comment](https://www.w3.org/TR/xml/#dt-comment), a [processing instruction]
> (https://www.w3.org/TR/xml/#dt-pi), or a [CDATA section](https://www.w3.org/TR/xml/#dt-cdsection). If they are needed 
> elsewhere, they MUST be [escaped](https://www.w3.org/TR/xml/#dt-escape) using either [numeric character references]
> (https://www.w3.org/TR/xml/#dt-charref) or the strings " &amp; " and " &lt; " respectively.

So while the error message is definitely unhelpful and I should fix that, the XML isn't well-formed.

But your first example uses CDATA, and it should work there, and it still fails. That's a real bug!

**Comment by faassen** [comment]
I've now released version 0.13.3 with the fix. Thank you again!

**Comment by dsrich** [comment]
Glad to help. The problem was with an ampersand in the CDATA section in my XML document, so thanks for fixing it
